### ISSUE TITLE
Executors and Placeholders refactoring

### ISSUE THREAD

**Issue by Ioloolo** [TriggerReactor/TriggerReactor#485]
What do you think?

**Comment by Sayakie** [comment]
As we said in https://github.com/TriggerReactor/TriggerReactor/pull/475#issuecomment-1055036236, it is a necessary task and inevitable. I couldn't do it because have no time, but I believe would be possible soon if we make a branch for that and push commits that worked on each other little by little.

**Comment by Sayakie** [comment]
Oops, didn't mean to close it.

**Comment by Ioloolo** [comment]
Is there any branch created yet?
Actually, I've been refactoring since yesterday.

**Comment by Sayakie** [comment]
No branches iirc yet

**Comment by Ioloolo** [comment]
@wysohn Could I create a new branch for refactorig?

**Comment by wysohn** [comment]
> @wysohn Could I create a new branch for refactorig?

Yes, please! Make sure to use `development` as the base branch

**Comment by Sayakie** [comment]
> **Note**
> Sponge API 8 was released in beta and they promised the API interface do not change. We should do divide Sponge project structure for 7 and 8 first if process Sponge also.

Here is a simple structure what I thought:
```mermaid
graph TD;
  TriggerReactorCore-->BukkitIntegratedAPI;
  TriggerReactorCore-->SpongeIntegratedAPI;
  BukkitIntegratedAPI-->id1[Legacy Bukkit];
  BukkitIntegratedAPI-->id2[Latest Bukkit];
  SpongeIntegratedAPI-->id3[Sponge API7];
  SpongeIntegratedAPI-->id4[Sponge API8];
```
/cc @Ioloolo

**Comment by Sayakie** [comment]
> **Note**
> Sponge API 8 was released in beta and they promised the API interface do not change. We should do divide Sponge project structure for 7 and 8 first if process Sponge also.

^^^
nvm seems to be not related to this issue and is good to create a new ticket. 😅

**Comment by rkdrns4747** [comment]
@Ioloolo Just quick question. Does this process resolves Issue #484 ?

**Comment by Ioloolo** [comment]
> @Ioloolo Just quick question. Does this process resolves Issue #484 ?

Yes. This PR will ensure operation for all EXs.

**Comment by rkdrns4747** [comment]
> > @Ioloolo Just quick question. Does this process resolves Issue #484 ?
> 
> Yes. This PR will ensure operation for all EXs.

Alright then, the Issue #484 will be closed and merged into this Issue #486

**Comment by Sayakie** [comment]
Resolved by #486
